mmv1/products/cloudrunv2/api.yaml

```diff
@@ -973,51 +973,74 @@ objects:
+                            properties:
+                              - !ruby/object:Api::Type::String
+                                name: name
+                                required: true
+                                description: |-
+                                  The header field name
+                              - !ruby/object:Api::Type::String
+                                name: value
+                                description: |-
+                                  The header field value
+                                default_value: ""
+                                send_empty_value: true
                     - !ruby/object:Api::Type::NestedObject
-                        name: "tcpSocket"
-                        description: |-
-                          TCPSocket specifies an action involving a TCP port. Exactly one of HTTPGet or TCPSocket must be specified.
-                        send_empty_value: true
-                        allow_empty_object: true
-                        # exactly_one_of:
-                        #   - template.0.containers.0.livenessProbe.0.httpGet
-                        #   - template.0.containers.0.livenessProbe.0.tcpSocket
-                        properties:
-                          - !ruby/object:Api::Type::Integer
-                            name: port
-                            description: |-
-                              Port number to access on the container. Must be in the range 1 to 65535. If not specified, defaults to 8080.
+                      name: "tcpSocket"
+                      description: |-
+                        TCPSocket specifies an action involving a TCP port. Exactly one of HTTPGet or TCPSocket must be specified.
+                      send_empty_value: true
+                      allow_empty_object: true
+                      # exactly_one_of:
+                      #   - template.0.containers.0.livenessProbe.0.httpGet
+                      #   - template.0.containers.0.livenessProbe.0.tcpSocket
+                      #   - template.0.containers.0.livenessProbe.0.grpc
+                      properties:
+                        - !ruby/object:Api::Type::Integer
+                          name: port
+                          description: |-
+                            Port number to access on the container. Must be in the range 1 to 65535. If not specified, defaults to 8080.
+                    - !ruby/object:Api::Type::NestedObject
+                      name: grpc
+                      description: |-
+                        GRPC specifies an action involving a GRPC port.
+                      send_empty_value: true
+                      allow_empty_object: true
+                      # exactly_one_of:
+                      #   - template.0.containers.0.livenessProbe.0.httpGet
+                      #   - template.0.containers.0.livenessProbe.0.tcpSocket
+                      #   - template.0.containers.0.livenessProbe.0.grpc
```

Comment:
commented out code... can be omitted.

Reply:
Sure! So this brings up the topic of `exactly_one_of` and what these commented out blocks were originally trying to ensure. 

`exactly_one_of` allows terraform to enforce that at least one of the given values are set. Unfortunately `exactly_one_of` and its sisters (`at_least_one_of` and `conflicts_with`) don't work under nested, itterable fields.

https://github.com/hashicorp/terraform-plugin-sdk/issues/470

The intention of these comments is to leave room for this behavior to be enabled at some future point in time although I don't see this issue being resolve in the near to moderate future. Thus I would be in favor of omitting these comments in future additions. It reduces code clarity with little value in my opinion.

Reply:
Given the lack of overall guidance (https://github.com/hashicorp/terraform-provider-google/issues/12648) on whether or not we should leave `exactly_one_of` and so on under repeatable nested fields as a comment, I will follow your advice and omit these comment lines at least here. 
There are a bit too many `exactly_one_of` comments in YAML files.